source/CMakeLists.txt

```diff
@@ -1,11 +1,9 @@
 add_library( freertos_plus_tcp STATIC )
 
-set_property(TARGET freertos_plus_tcp PROPERTY C_STANDARD 90)
+set_property(TARGET freertos_plus_tcp PROPERTY C_STANDARD 99)
```

Comment:
@tony-josi-aws @amazonKamath
Do we make any claims that FreeRTOS+TCP is C90 Compliant? If so can we add this claim to the [README.md](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP#freertos-plus-tcp-library) on this repo, and the [FreeRTOS+TCP](https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/index.html) landing page?

If we don't make any claims about C90 compliance I see no reason not to accept this change to C99 here.

Reply:
I wasn't sure about this either, however there are a few things that aren't C90 compliant like unnamed unions and commas at the end of enums.

Reply:
@HTRamsey, Kernel still remains at C90.  There was debate sometime last year, but we decided to remain at C90.  For the TCP stack, it seems like the Source/* is at C90 and with pedantic flag enabled. We should be able to catch if there is some non-compliant code, however it looks like that is not the case for Portable/*.   My colleague @tony-josi-aws  is investigating and we will get back to you soon.  Thank you.

Reply:
@HTRamsey  FYI, We identified code which has creeped in over the last year(s) which are not C90 compliant.  However, we are trying to figure out what is preventing the build system from throwing the error even though the pedantic flag is set and the standard points to C90 in Cmake file.   Did you notice compile warnings/error for unnamed unions? 
Thanks!

Reply:
@amazonKamath yeah they're from struct xNetworkEndPoint

Reply:
@HTRamsey  did the compiler complain about it for you?

Reply:
@HTRamsey, the CI build checks were failing to report C90 incompatible code in the TCP library header files due to a [CMake configuration (SYSTEM keyword)](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/826e261a50b2296d8c79587a3927884fc401dcd4/source/CMakeLists.txt#L93) that was set. As we have identified what is causing the build checks to not enforce the C90 standard, we are finding the best way to remove the incompatibility without any breaking changes.

Reply:
[tcp_test.txt](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/files/14122577/tcp_test.txt)